pkg/apis/core/helper/helpers.go

```diff
@@ -365,6 +365,16 @@ func SystemComponentsAllowed(worker *core.Worker) bool {
 	return worker.SystemComponents == nil || worker.SystemComponents.Allow
 }
 
+// GetResourceByName returns the first NamedResourceReference with the given name in the given slice, or nil if not found.
```

Comment:
We don't need to write `first`, resource names are unique [ref](https://github.com/gardener/gardener/blob/8da80d8eb5b39f9104bd9049f72907c55a472748/pkg/apis/core/validation/shoot.go#L793-L807).
```suggestion
// GetResourceByName returns the NamedResourceReference with the given name in the given slice, or nil if not found.
```